File: android/terminalSdk/src/main/java/io/nearpay/terminalsdk/data/dto/Merchant.kt
package io.nearpay.terminalsdk.data.dto

import kotlinx.serialization.SerialName
import kotlinx.serialization.Serializable

@Serializable
data class Merchant(
    @SerialName("id")
    val id: String,

    @SerialName("name")
    val name: LanguageContent,

    @SerialName("address")
    val address: LanguageContent,

    @SerialName("category_code")
    val categoryCode: String
)

File: lib/models/data/merchant.dart
import 'language_content.dart';

class Merchant {
  final String id;
  final LanguageContent name;
  final LanguageContent address;
  final String categoryCode;

  Merchant({
    required this.id,
    required this.name,
    required this.address,
    required this.categoryCode,
  });

  factory Merchant.fromJson(Map<String, dynamic> json) {
    return Merchant(
      id: json['id'] as String,
      name: LanguageContent.fromJson(json['name'] as Map<String, dynamic>),
      address: LanguageContent.fromJson(json['address'] as Map<String, dynamic>),
      categoryCode: json['categoryCode'] as String,
    );
  }

  Map<String, dynamic> toJson() {
    return {
      'id': id,
      'name': name.toJson(),
      'address': address.toJson(),
      'categoryCode': categoryCode,
    };
  }
}
